conn.go

```diff
@@ -125,6 +131,11 @@ func NewConn(ctx context.Context, conn net.Conn, opts *ConnOptions) (*Conn, erro
 	return c, nil
 }
 
+type CloseEvent struct {
```

Comment:
I think the close event should return a `ConnError` and have the same semantics as `Conn.Close` so that error handling is uniform regardless of where the error originates.